### PR TITLE
Coordinate transformations

### PR DESCRIPTION
**Pull request by zichunhao**
Implemented coordinates transformations:
- For massive particles: (E, px, py, pz) <-> (E, pT, y, phi), where y is the rapidity.
- For massless particles: (E, px, py, pz) <-> (E, pT, eta, phi), where eta is the pseudorapidity.
- Transformations to relative coordinates (pT_rel, eta_rel, phi_rel) are implemented from (E, px, py, pz) and (E, pT, eta, phi).
- Transformation to absolute coordinates (E, px, py, pz) and (E, pT, eta, phi) are implemented from (E, pT, eta, phi) and jet momenta (in either polar or Cartesian coordinates).

### CODE REVIEW

**Comment by rkansal47** [comment]
Awesome, this looks great!

Couple things:
 - Let's stick to pseudorapidity only, even for massive particles, to stay consistent with [coffea](https://coffeateam.github.io/coffea/api/coffea.nanoevents.methods.vector.LorentzVector.html#coffea.nanoevents.methods.vector.LorentzVector).
 - We should also add tests for these at some point (but I can merge this for now without them)

**Comment by zichunhao** [comment]
Should we 
1. still keep the ones with rapidities and change the name `p4_polar_from_p4_cartesian_massless` to `p4_polar_from_p4_cartesian`, and what was `p4_polar_from_p4_cartesian` to `p4_polar_from_p4_cartesian_massive`, or
2. Simply delete the transformations with rapidity?

**Comment by rkansal47** [comment]
Good point, it doesn't hurt to keep the rapidity transformation as well.

I would just suggest a name change then - where instead of `p4_polar` we use `PtEtaPhiE` or `PtYPhiE`, for pseudorapidity and rapidity respectively, etc. to avoid any confusion (and also to be more consistent with Coffea https://coffeateam.github.io/coffea/api/coffea.nanoevents.methods.vector.PtEtaPhiMLorentzVector.html).

File: jetnet/utils/coord_transform.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+    # (E/c, p_x, p_y, p_z) -> (E/c, p_T, y, phi)
+    p0, px, py, pz = __unbind(p4, axis=-1)
+    pt = __sqrt(px**2 + py**2)
+    y = 0.5 * __log((p0 + pz + eps) / (p0 - pz + eps))
+    phi = __arctan2(py, px)
+
+    return __stack([p0, pt, y, phi], axis=-1)
+
+
+def p4_cartesian_from_p4_polar(
```
Comment (rkansal47) [inline jetnet/utils/coord_transform.py:85]: -> `YPhiPtE_to_cartesian`

File: jetnet/utils/coord_transform.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+    p0, pt, y, phi = __unbind(p4, axis=-1)
+    px = pt * __cos(phi)
+    py = pt * __sin(phi)
+    # get pz
+    mt = p0 / (__cosh(y) + eps)  # get transverse mass
+    pz = mt * __sinh(y)
+    return __stack([p0, px, py, pz], axis=-1)
+
+
+def polarrel_from_p4_cartesian(
```
Comment (rkansal47) [inline jetnet/utils/coord_transform.py:114]: -> `cartesian_to_relEtaPhiPt`

**Review by rkansal47 — APPROVED** [review]

**Pull request merged**